libc/stdlib_alloc.c

```diff
@@ -0,0 +1,318 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libc-tests
+ *
+ * TESTED:
+ * malloc, calloc,
+ * realloc, free
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Mateusz Bloch
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdlib.h>
+#include <malloc.h>
+#include <string.h>
+
+#include <unity_fixture.h>
+
+#define BLOCK_SIZE 100
+#define LARGE_SIZE 100
+
+/*
+Tets group for:
+malloc, calloc,
+realloc, free
+*/
+
+TEST_GROUP(stdlib_malloc);
+
+TEST_SETUP(stdlib_malloc)
+{
+}
+
+
+TEST_TEAR_DOWN(stdlib_malloc)
+{
+}
+
+TEST(stdlib_malloc, malloc_basic)
+{
+	void *ptr;
+	ptr = malloc(BLOCK_SIZE);
+	TEST_ASSERT_NOT_NULL(ptr);
+#ifndef phoenix  // Temporarily disabled on phoenix because issues: #660
+	TEST_ASSERT_GREATER_OR_EQUAL_INT(BLOCK_SIZE, malloc_usable_size(ptr));
+#endif
+	memset(ptr, 0x5a, BLOCK_SIZE);
+	TEST_ASSERT_EACH_EQUAL_HEX8(0x5a, ptr, BLOCK_SIZE);
+	free(ptr);
+}
+
+TEST(stdlib_malloc, malloc_small)
+{
+	void *ptr;
+	ptr = malloc(1);
+	TEST_ASSERT_NOT_NULL(ptr);
+#ifndef phoenix
+	TEST_ASSERT_GREATER_OR_EQUAL_INT(1, malloc_usable_size(ptr));
+#endif
+	memset(ptr, 0x2d, 1);
+	TEST_ASSERT_EACH_EQUAL_HEX8(0x2d, ptr, 1);
+	free(ptr);
+}
+
+TEST(stdlib_malloc, malloc_large)
+{
+	void *ptr;
+	ptr = malloc(BLOCK_SIZE * LARGE_SIZE);
+	TEST_ASSERT_NOT_NULL(ptr);
+#ifndef phoenix
+	TEST_ASSERT_GREATER_OR_EQUAL_INT(BLOCK_SIZE * LARGE_SIZE, malloc_usable_size(ptr));
+#endif
+	memset(ptr, 0xb4, BLOCK_SIZE * LARGE_SIZE);
+	TEST_ASSERT_EACH_EQUAL_HEX8(0xb4, ptr, BLOCK_SIZE * LARGE_SIZE);
+	free(ptr);
+}
+
+TEST(stdlib_malloc, malloc_multiple)
+{
+	void *ptr0;
+	ptr0 = malloc(sizeof(int));
+	TEST_ASSERT_NOT_NULL(ptr0);
+	TEST_ASSERT_TRUE((uintptr_t)ptr0 % sizeof(int) == 0);
```

Comment:
I suggest using `TEST_ASSERT_EQ` (or EQUAL if EQ doesn't exist)
```suggestion
	TEST_ASSERT_EQ((uintptr_t)ptr0 % sizeof(int),  0);
```